### ISSUE TITLE
Lattest commits broke everything

### ISSUE THREAD

**Issue by alkeryn** [nixos-rocm/nixos-rocm#18]
I get nix evals issue since your latest commit and can't build my system, i had to disable that overlay to build

**Comment by acowley** [comment]
Oh no, I’m sorry! Do you have any error message you can share?

**Comment by alkeryn** [comment]
Oh it is okay, you don't have to apologize ^
it was kind of an issue to find the culprit but that's all i got
![image](https://user-images.githubusercontent.com/11599075/66966882-19732480-f07f-11e9-9c1c-a92427e213a8.png)
![image](https://user-images.githubusercontent.com/11599075/66966887-1bd57e80-f07f-11e9-9773-be0b414c67d9.png)
my overlays look like that : 
![image](https://user-images.githubusercontent.com/11599075/66966900-25f77d00-f07f-11e9-9e16-aa78f73c30a5.png)

**Comment by alkeryn** [comment]
But basically most nix commands weren't working (as i got that file both imported as an overlay in my configuration.nix and in .config/nixpkgs/overlays.nix) ^

**Comment by acowley** [comment]
Shoot, I'm stumped at the moment. I'd definitely appreciate if you can help bisect this by commenting out parts of the `nixos-rocm` `all-packages.nix`, but that would require you have a local checkout of the repo rather than using `fetchGit` directly.

I've tried running `nix-build --check` on things, but it's not showing any problems for me. This definitely cranks up the pressure on figuring out CI for this overlay.

**Comment by alkeryn** [comment]
@acowley It is okay i wasn't really using it much anyway
was mostly using hip tools
i might look into what commit specifically broke it ^

**Comment by alkeryn** [comment]
hey @acowley i found the lattest working commit:
https://github.com/nixos-rocm/nixos-rocm/commit/288bb21520f07d63ce8d7ff82942526beda7d761
so it seems that this commit broke things : 
https://github.com/nixos-rocm/nixos-rocm/commit/e4ef8ab0a03c664023800be696a6b11c0bc99803
what is weird tho is that it used to be working but since i'm using a clone nixos/nixpkgs/master as my nixpkgs i have the lattest updates so maybe a recent thing with it broke compatibility, tho i doubt it since i tried building like 400 commit in the past.

**Comment by acowley** [comment]
Oh! Did you see the note with the siren emojis in the README? The overlay now requires a hardware argument to avoid the builders needing a lot of permissions. Upstream ROCm changed things so that enumerating the installed hardware requires more of the users performing the builds, so my choices where to ask users to do adjust their `nixbld` users (which is hard because there are dozens of them), or to not enumerate installed hardware at build-time. I went with the latter, which requires that you pass a GPU IP identifier (e.g. `"gfx803"` for Polaris).

**Comment by alkeryn** [comment]
@acowley Yeah that was it, it fixed it, thanks :)
sorry for not noticing it sooner
also do you have support for nvidia gpu's as i know hip can be compiled for nvidia gpu too ^
Also where do you find those gpu ip identifier ?

**Comment by acowley** [comment]
I don't know the answer to either of those questions! What I do for my own code is write `CMakeLists.txt` files that run `hipify-perl` over `.cu` files, and then use the `hip` tools to build things, while using `nvcc` if I'm not in AMD mode based on a `cmake` flag. I went this route so as to not ruffle the feathers of Nvidia-only people by introducing something that doesn't help them. On AMD machines, I have hip, and on Nvidia machines I just have the CUDA tooling.

As for the GPU identifier, I don't understand why it's so hard to find a reference, but your question prompted me to search again and I finally found [this](https://www.llvm.org/docs/AMDGPUUsage.html#processors) which looks pretty good. But one does have to bear in mind that ROCm support is extremely limited; I think 803, 900, and 906 are the only targets currently supported.
